### ISSUE TITLE
Q10914 Horde Terrokar  Persuit Mobs not attackable when flying - endless circleling

### ISSUE THREAD

**Issue by marrvin2** [EchoTiger/NewQuestingPack#7]
**Update:**
*sorry*  this seems to be a general problem. (encaoutered again on Q10027)
_In my interface option I did not check  "Auto dismount in flight" (which is default off)._
If this is not enabled,  even hostile mobs are flagged as not attackable when flying (no  "sword" mouse cursor).

**you should enable this cVar when questing or you will have a lot of issues** :D

------- old "ticket" text -------------

as long as the char is in the air the mobs are not attackable:

Flying from <-3596.882, 5179.412, 19.9388> to Type: Kill, Name: Auchenai Initiate, NavType: Fly
[Flightor] Set up for movement from <-3596.882, 5179.412, 19.9388> to <-3605.65, 5204.021, -20.27249>
Clearing POI: POI is not attackable

I changed the code to

		<Code File="ForcedDismount" />
		<DefaultNavType Value="Run" />
		<Code File="KillUntilComplete" QuestId="10914" MovementBy="NavigatorOnly" >
			<HuntingGrounds>
				<Hotspot X="-3664.344" Y="5135.707" Z="-22.5169" />
				<Hotspot X="-3576.688" Y="5268.199" Z="-20.34017" />
				<Hotspot X="-3436.809" Y="5254.816" Z="-20.81991" />
			</HuntingGrounds>
			<PursuitList>
				<PursueUnit Id="21284" PursueWhen="!IsObjectiveComplete(1, 10914)" />
				<PursueUnit Id="21285" PursueWhen="!IsObjectiveComplete(2, 10914)" />
			</PursuitList>
		</Code>
		<DefaultNavType Value="Fly" />

to fix the issue.
The HuntingGrounds is right next to the Questgiver, so dismounting after Quest PickUp is no issue.

**Comment by EchoTiger** [comment]
This shouldn't be an issue in Honorbuddy as Honorbuddy calls a forced dismount behavior before attempting to engage the NPC.
This forced dismount behavior must see "complete" before the engage code is ran.

Something is happening on your installation that is messing with the forced dismount code.
Either way, I've added this automatic enabling of this to prevent potential issues such as the one you're having.

**Comment by marrvin2** [comment]
Hi,
to counter any outside issues, I did take precautions before running your new profiles:
All Ingame Addons are disabled  and I am using a "clean" HB Key that as no plugins enabled.
The Honorbuddy installation is "unmodified" (no extra plugins).
When I look at the log  and at the char  ingame I saw that  the Bot POI was cleared BEFORE the mob was enganged (as it was to far away) due to its  "unattackable"-state.
I attached a log excerp from killing Nagas in Zangamarsh  if somebody from hb staff wants to investigate it.  just search for  "POI is not attackable".

[cannot attack.log.txt](https://github.com/EchoTiger/NewQuestingPack/files/1000901/cannot.attack.log.txt)

hope this helps.

.
